pkg/volume/volume.go

```diff
@@ -132,11 +132,39 @@ type Attacher interface {
 	Attach() error
 }
 
+type PhysicalAttacher interface {
+	// WaitForAttach examines the file system for the device; if
+	// found that means the disk is attached and this function
+	// returns otherwise it polls until the device appears. If the
+	// device is not attached within the timeout period an error
+	// is returned.
+	WaitForAttach() error
```

Comment:
Also, consider having this method return the path for the newly attached device (i.e. `makeGlobalPDName(...)` for GCE so that the calling code can use it for the bind mount step.